includes/admin/reports/class-wc-admin-report.php

```diff
@@ -513,7 +513,13 @@ public function calculate_current_range( $current_range ) {
 		switch ( $current_range ) {
 
 			case 'custom' :
-				$this->start_date = strtotime( sanitize_text_field( $_GET['start_date'] ) );
+
+				if ( ! isset( $_GET['wc_reports_nonce'] ) || ! wp_verify_nonce( $_GET['wc_reports_nonce'], 'custom_range' ) ) {
+					wp_safe_redirect( remove_query_arg( array( 'start_date', 'end_date', 'range', 'wc_reports_nonce' ) ) );
+					exit;
+				}
+
+				$this->start_date = max( 1262304000, strtotime( sanitize_text_field( $_GET['start_date'] ) ) );
```

Comment:
I realize WC came out after this, but what about the possible use case of needing to run a report on data that was imported and existed before 2010? Definitely doesn't seem common -- but seems possible.